python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -54,19 +54,67 @@
 from dagster._core.types.dagster_type import DagsterType
 from dagster._utils.forked_pdb import ForkedPdb
 from dagster._utils.merger import merge_dicts
+from dagster._utils.warnings import deprecation_warning
 
 from .compute import OpExecutionContext
 from .system import StepExecutionContext, TypeCheckContext
 
 
 def _property_msg(prop_name: str, method_name: str) -> str:
     return (
-        f"The {prop_name} {method_name} is not set on the context when a solid is directly invoked."
+        f"The {prop_name} {method_name} is not set on the context when an op is directly invoked."
     )
 
 
-class UnboundOpExecutionContext(OpExecutionContext):
-    """The ``context`` object available as the first argument to a solid's compute function when
+class BoundProperties:
+    def __init__(
+        self,
+        op_def: OpDefinition,
+        tags: Mapping[Any, Any],
+        hook_defs: Optional[AbstractSet[HookDefinition]],
+        alias: str,
+        assets_def: Optional[AssetsDefinition],
+        resources: Resources,
+        op_config: Any,
+    ):
+        """Maintains the properties of the context that are provided at bind time.
+        This class is not implemented as a NamedTuple because requires_typed_event_stream
+        and type_event_stream_error_message should be mutable.
+        """
+        self.op_def = check.inst_param(op_def, "op_def", OpDefinition)
+        self.tags = check.dict_param(tags, "tags")
+        self.hook_defs = check.opt_set_param(hook_defs, "hook_defs", HookDefinition)
+        self.alias = check.str_param(alias, "alias")
+        self.assets_def = check.opt_inst_param(assets_def, "assets_def", AssetsDefinition)
+        self.resources = check.inst_param(resources, "resources", Resources)
+        self.op_config = op_config
+        self.requires_typed_event_stream = False
+        self.typed_event_stream_error_message = None
```

Comment:
nit: use a frozen dataclass or NamedTuple, something immutable

Reply:
```
        self.requires_typed_event_stream = False
        self.typed_event_stream_error_message = None
```
I think these two are more execution related than bind time

Reply:
I moved those to `InvocationProperties`. That meant that `InvocationProperties` needed to not be a `NamedTuple` anymore so that `requires_typed_event_stream` and `typed_event_stream_error_message` could be updated. That seems more ok to me since most of the attributes of `InvocationProperties` can be mutated. I could be more defensive about how the properties can get mutated though if we're worried about it